src/Components/test/testassets/BasicTestApp/RouterTest/TestRouterWithLazyAssembly.razor

```diff
@@ -0,0 +1,74 @@
+@using Microsoft.AspNetCore.Components.Routing
+@using Microsoft.AspNetCore.Components.WebAssembly
+@using System.Linq
+@using System.Reflection
+
+@inject WebAssemblyLazyAssemblyLoader lazyLoader
+
+<Router AppAssembly="@typeof(BasicTestApp.Program).Assembly" AdditionalAssemblies="@lazyLoadedAssemblies" OnNavigate="@OnNavigate">
+    <Found Context="routeData">
+        @if (loading) {
+            <div style="padding: 20px;background-color:blue;color:white;" id="loading-banner" @onclick="HideLoadingBar">
+                <p>Loading the requested page...</p>
+            </div>
+        } else {
+            <RouteView RouteData="@routeData" />
+        }
```

Comment:
How does this work if I want to preserve the current view on the page until I have loaded all the remaining assemblies to switch to a new page?

Reply:
That seems like an unusual UX, and maybe confusing. You’d want to show something at least to indicate that your click is acknowledged. 

I think it’s fine if the natural flow is going to a “loading” screen. In the case where a developer really doesn’t want to, they can accept the navigation gesture as a button click or link with no href, then inside the handler start lazy loading assemblies while showing some UI for it, and then programmatically navigate to the new URL.